packages/roosterjs-editor-dom/lib/utils/normalizeRect.ts

```diff
@@ -0,0 +1,17 @@
+import { Rect } from 'roosterjs-editor-types';
+
+/**
+ * A ClientRect of all 0 is possible. i.e. chrome returns a ClientRect of 0 when the cursor is on an empty p
+ * We validate that and only return a rect when the passed in ClientRect is valid
+ */
+export default function normalizeRect(clientRect: ClientRect): Rect {
+    let { left, right, top, bottom } = clientRect || <ClientRect>{};

```

Comment:
suggestion/nit: I think the `<ClientRect>{}` fallback will make the left, right, top, bottom all as undefined and which will become `NaN` when added. Maybe it should be `<ClientRect>{ left: 0, right: 0, top: 0, bottom: 0 }`

Reply:
Technically this is "OK" as `nan > x` will return false.

Reply:
I'll change it to create more clarity around what's happening here

Reply:
Might